client/components/Dashboard.tsx

```diff
@@ -31,47 +29,243 @@ const GET_PROJECT = gql`
+        complexity: false,
+    });
+    const [order, setOrder] = useState<boolean>(); // ascending vs descending on click
+    const [arrow, setArrow] = useState<ISState['arrow']>({
+        // adjusting arrow based on ascending vs descending
+        time: '',
+        depth: '',
+        complexity: '',
+    });
+    // toggle classes on click in the Queries component
+    const setToggle = (arg: string) => {
+        if (arg === 'time') {
+            setStyle({
+                ...style,
+                time: true,
+                depth: false,
+                complexity: false,
+            });
+        } else if (arg === 'depth') {
+            setStyle({
+                ...style,
+                time: false,
+                depth: true,
+                complexity: false,
+            });
+        } else if (arg === 'complexity') {
+            setStyle({
+                ...style,
+                time: false,
+                depth: false,
+                complexity: true,
+            });
+        }
+    };
+    // graphql calling
     const { data: dataR, loading: loadingR } = useQuery(GET_USER_DATA, {
         variables: {
-            userId: '6286978e12716d47e6884194',
+            userId: '62ba5b743f40d18829e018a1',
         },
     });
     const { data, loading } = useQuery(GET_PROJECT, {
         variables: {
-            projectId: '628e864e76cdbbec53f36010',
+            projectId: '62ba5bd43f40d18829e018c4',
         },
     });
 
-    const [projects, setProjects] = useState<Projects['projects']>();
+    // const [projects, setProjects] = useState<Projects['projects']>();   // commented out due to still not knowing how you guys want our project selection to look like. Waiting for instructions
     const [selectedProject, selectProject] = useState<SelectedProject['project']>();
+    const [queries, selectQuerries] = useState<ProjectQuery[]>();
 
-    const test = (pr: any): void => {
-        selectProject(pr);
-    };
+    // const test = (pr: any): void => { // this is poorly named fuction that selects the project
+    //     selectProject(pr);
+    // };
 
-    useEffect(() => {
-        if (!loadingR && dataR) {
-            setProjects(dataR.user.projects);
+    // sorting functions
+
+    const sortByTime = (): any => {
+        if (order === true) {
+            setOrder(false);
+            // setArrow();
+            setArrow({
+                ...arrow,
+                time: '↑',
+                depth: '',
+                complexity: '',
+            });
+        } else {
+            setOrder(true);
+            setArrow({
+                ...arrow,
+                time: '↓',
+                depth: '',
+                complexity: '',
+            });
+        }
+        if (order) {
+            const newArr = [];
+            if (selectedProject) {
+                // eslint-disable-next-line no-restricted-syntax, guard-for-in
+                for (const key in selectedProject?.queries) {
```

Comment:
Can the `for...in` loop be replaced with `map()`? It also doesn't seem necessary to copy `newArr` before sorting since it was just initialized.

Reply:
.map()? Have you checked the typeof for queries? I think they are not arrays